cmd/deploy/deploy.go

```diff
@@ -92,26 +91,24 @@ func Deploy(ctx context.Context) *cobra.Command {
 			// This is needed because the deploy command needs the original kubeconfig configuration even in the execution within another
 			// deploy command. If not, we could be proxying a proxy and we would be applying the incorrect deployed-by label
 			os.Setenv(model.OktetoWithinDeployCommandContextEnvVar, "false")
-			if okteto.IsOkteto() && options.Namespace != "" {
-				create, err := utils.ShouldCreateNamespace(ctx, options.Namespace)
+
+			if err := contextCMD.LoadManifestV2WithContext(ctx, options.Namespace, options.ManifestPath); err != nil {
+				return err
+			}
+
+			if okteto.IsOkteto() {
+				create, err := utils.ShouldCreateNamespace(ctx, okteto.Context().Namespace)
```

Comment:
`SouldCreateNamespace` and `ExecuteCreateNamespace` are in different packages?

Reply:
Yes, `ShouldCreateNamespace` should be in a different package. I created an issue to track it:
https://github.com/okteto/okteto/issues/2112